src/Runner.Worker/ExecutionContext.cs

```diff
@@ -593,7 +602,31 @@ public void SetTimeout(TimeSpan? timeout)
             if (timeout != null)
             {
                 _cancellationTokenSource.CancelAfter(timeout.Value);
+                m_timeoutSetAt = DateTime.UtcNow;
```

Comment:
`m_timeoutSetAt` - perhaps `m_timeoutStartedAt`?

Reply:
Just `m_startedAt`? so that `elapsed = DateTime.UtcNow - m_startedAt.Value`

Reply:
I'd rather keep it `m_timeoutStartedAt` to clearly imply it's for timeout related stuff, unlike `_record.StartedAt`